### DISCUSSION TITLE
Sonoff ZBBridge Zigbee module error

### DISCUSSION THREAD

**Discussion by Malakai-DB** [arendst/Tasmota#13261] [Q&A] [answered]
Hello,

I recently bought a Sonoff ZBBridge that I successfully flashed with tasmota-zbbridge.bin firmware through serial.
After the reboot and setting the wifi credentials I was able to connect to the Web UI.
Now when I try to flash the Zigbee module like instructed [here](https://zigbee.blakadder.com/Sonoff_ZBBridge.html), I get an error in the console (weblog set to 3) : `XMD: SYNC timeout`.

Here is the entire output of the console : 
```
00:00:00.002 HDW: ESP8266EX
00:00:00.031 CFG: Loaded from flash at FD, Count 21
00:00:00.034 CFG: CR 357/699, Busy 0
00:00:00.036 ZIG: ZBBridge EEPROM found at address 0x50
00:00:00.037 SNS: Hardware Serial
00:00:00.038 SRC: Restart
00:00:00.039 Project tasmota Tasmota Version 9.5.0(zbbridge)-2_7_4_9(2021-06-17T08:27:09)
00:00:00.040 ZIG: Resetting EZSP device
00:00:00.189 ZbEZSPReceived: time-out, discarding F8
00:00:00.190 WIF: Attempting connection...
00:00:00.526 WIF: Connecting to AP1 xxx Channel 11 BSSId xx:xx:xx:xx:xx:xx in mode 11n as tasmota_9F51C6-4550...
00:00:01.753 WIF: Attempting connection...
00:00:02.754 WIF: Attempting connection...
00:00:03.756 WIF: Connected
00:00:04.006 HTP: Web server active on tasmota_9F51C6-4550 with IP address 192.168.xx.xx
00:00:04.706 RTC: UTC 2021-09-29T13:10:23, DST 2021-03-28T02:00:00, STD 2021-10-31T03:00:00
14:10:24.009 RSL: INFO1 = {"Info1":{"Module":"Sonoff ZbBridge","Version":"9.5.0(zbbridge)","FallbackTopic":"cmnd/DVES_9F51C6_fb/","GroupTopic":"cmnd/tasmotas/"}}
14:10:24.010 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota_9F51C6-4550","IPAddress":"192.168.xx.xx"}}
14:10:24.011 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart"}}
14:10:27.457 APP: Boot Count 6
14:10:27.569 CFG: Saved to flash at FC, Count 22, Bytes 4096
14:10:28.425 RSL: STATE = {"Time":"2021-09-29T14:10:28","Uptime":"0T00:00:11","UptimeSec":11,"Vcc":3.447,"Heap":31,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"xxx","BSSId":"xx:xx:xx:xx:xx:xx","Channel":11,"Mode":"11n","RSSI":100,"Signal":-38,"LinkCount":1,"Downtime":"0T00:00:05"}}
14:10:34.041 ZIG: Resetting EZSP device
14:10:34.328 HTP: Information
14:10:34.476 ZbEZSPReceived: time-out, discarding FE0D0180020000000001010878
14:10:39.220 ZIG: timeout, goto label 99
14:10:39.222 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
14:10:39.222 ZIG: Abort
14:10:39.223 ZIG: Stopping (99)
14:10:42.270 HTP: Main Menu
14:10:55.937 HTP: Firmware Upgrade
14:11:02.027 CFG: Saved to flash at 1FB, Count 23, Bytes 4096
14:11:02.027 UPL: File ncp-uart-sw_6.7.8_115200.ota
14:11:03.013 UPL: File type 5
14:11:05.257 UPL: Progress 100 kB
14:11:07.388 UPP: Multicast (re)joined
14:11:07.389 UPL: Transfer 190756 bytes
14:11:07.389 UPL: Successful 190756 bytes
14:11:07.400 HTP: Upload transfer
14:11:07.439 XMD: Init bootloader
14:11:07.439 ZIG: Bootloader pin 5
14:11:08.659 XMD: Init sync
14:11:38.662 XMD: SYNC timeout
14:11:38.663 XMD: Failed
14:11:38.663 ZIG: Bootloader pin 5
14:11:39.036 HTP: Upload done
```

Like it is mentioned in the console logs, I am on Tasmota 9.5.0, but I also tried the following versions with the same result : 9.1 and 8.5.1
I also tried for each of Tasmota version the following Zigbee module firmware : 
- ncp-uart-sw-6.7.8_115200.ota
- ncp-uart-nsw_6.7.9_115200.ota
- ncp-uart-sw_6.7.6_115200.ota

What am I doing wrong?
Has anyone been able to flash the Zigbee module on any of Tasmota versions I have been trying to use?
If any other information is necessary, please just ask and I will provide it.

**Comment by Malakai-DB** [comment, marked as answer]
Today I made a reset to the ZBBridge (reset 1) and only configured the WiFi and the Weblog to 4.
After this I tried to upload ncp-uart-sw-6.7.8_115200.ota using 3 different browsers (Firefox, Chrome and Edge) and I still got the same result : 
```
XMD: SYNC timeout
XMD: Failed
```
Also tried 3 different power plugs just in case...

I can't understand why it doesn't work and what exactly is the `SYNC timeout` meaning.
Is the Sonoff ZBBridge not supported anymore by Tasmota?

**Reply by sfromis** [reply]
Nothing has gone out of support, many are using the device. On a general level, `XMD: SYNC timeout` means trouble communicating to the Zigbee chip (using XModem) during firmware upload. This should not at all depend on which browser you use. Unless hardware has changed (or is damaged), it could plausibly be due to issues with wiring (including the potential of solder bridges, torn pads/traces, or such) or "power quality issues". Does not look like a Tasmota issue, but obviously I can't make a clear diagnosis.

**Reply by Malakai-DB** [reply]
Thanks @sfromis for your answer, especially this part : "Unless hardware has changed (or is damaged), it could plausibly be due to issues with wiring (including the potential of solder bridges, torn pads/traces, or such)". The `XMD: SYNC timeout` error wasn't clear for me.
This could be it, as I am quite new in the soldering world and I might have damaged the PCB in a way or another when trying to flash Tasmota on the device.
Now that I know the device should correctly work with Tasmota and the problem might come from my bad skills at soldering, I will try to order a new unit and be more careful when doing/undoing the soldering.

**Reply by sfromis** [reply]
You could also try inspecting the soldering (and wiring) you've done closely, looking out for suspicious details. Especially around TX/RX.

**Comment by Malakai-DB** [comment]
I just wanted to report that it was indeed a hardware issue.
I received another unit that I flashed with Tasmota and was able to successfully upload the Zigbee firmware.
So anyone who may have this kind of issue, the problem is hardware (mine was damaged during soldering).
Thanks @sfromis again for the help.

**Comment by zivojinov** [comment]
I had the same issue, then I read documentation and this point helps me:

When the ZbBridge is flashed with Tasmota, disconnect all breadboard wires and power the board using its USB port with a 5V 1A power supply.

**Comment by aTommyC** [comment]
For anyone reading this still, I just had the same experience. 

Tried to upload the Zigbee chip firmware, but got stuck at "Upload transfer". 

Using Tasmota 14.05.0 (release zbbridge, module 75).
Tried ncp uart 6.7.9 and 7.4.5.0. Same result.

Switched from a USB charger unit to the USB port of my laptop and retried. 7.4.5.0 did not run on my ZBBridge, but 6.7.9 did.

Flashing ncp v7.x gave the following result:

11:09:25.804 ZIG: Resetting EZSP device
11:09:27.010 RSL: RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
11:09:27.011 ZIG: ZbEZSPSend 000008
11:09:27.063 ZIG: {"ZbEZSPReceived":"00000C021073"}
11:09:27.065 RSL: RESULT = {"ZbState":{"Status":55,"Version":"7.3.1.0","Protocol":12,"Stack":2}}
11:09:27.066 RSL: RESULT = {"ZbState":{"Status":98,"Message":"Only EZSP protocol v8 is currently supported"}}
11:09:27.068 RSL: RESULT = {"ZbState":{"Status":99,"Message":"Abort"}}
11:09:27.068 ZIG: Abort
11:09:27.069 ZIG: Stopping (99)

Also, reflashing required the firmware upload to begin before ZB was started, so I kept a couple of tabs opened, resetting the device, and started flashing ASAP at boot.

**Reply by sfromis** [reply]
Yeah, Tasmota does not support 7.4.5.0, use binaries from here:
https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp
